reactor-pool-micrometer/src/main/java/reactor/pool/introspection/micrometer/DocumentedPoolMeters.java

```diff
@@ -0,0 +1,205 @@
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.pool.introspection.micrometer;
+
+import io.micrometer.common.docs.KeyName;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.docs.DocumentedMeter;
+
+/**
+ * Meters used by {@link Micrometer} utility.
+ */
+enum DocumentedPoolMeters implements DocumentedMeter {
+
+	ACQUIRED {
+		@Override
+		public String getName() {
+			return "%s.resources.acquired";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.COUNTER;
+		}
+	},
+	ALLOCATED {
+		@Override
+		public String getName() {
+			return "%s.resources.allocated";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.COUNTER;
+		}
+	},
+	IDLE {
+		@Override
+		public String getName() {
+			return "%s.resources.idle";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.COUNTER;
+		}
+	},
+	PENDING_ACQUIRE {
+		@Override
+		public String getName() {
+			return "%s.resources.pendingAcquire";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.COUNTER;
+		}
+	},
+
+
+	ALLOCATION {
+		@Override
+		public String getName() {
+			return "%s.allocation";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.TIMER;
+		}
+
+		@Override
+		public KeyName[] getKeyNames() {
+			return AllocationTags.values();
+		}
+	},
+
+	DESTROYED {
+		@Override
+		public String getName() {
+			return "%s.destroyed";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.DISTRIBUTION_SUMMARY;
+		}
+	},
+
+	SUMMARY_IDLENESS {
+		@Override
+		public String getName() {
+			return "%s.resources.summary.idleness";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.DISTRIBUTION_SUMMARY;
+		}
+	},
+
+	SUMMARY_LIFETIME {
+		@Override
+		public String getName() {
+			return "%s.resources.summary.lifetime";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.DISTRIBUTION_SUMMARY;
+		}
+	},
+
+	RECYCLED {
+		@Override
+		public String getName() {
+			return "%s.recycled";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.COUNTER;
+		}
+	},
+
+	RECYCLED_NOTABLE {
+		@Override
+		public String getName() {
+			return "%s.recycled.notable";
+		}
+
+		@Override
+		public Meter.Type getType() {
+			return Meter.Type.COUNTER;
+		}
+
+		@Override
+		public KeyName[] getKeyNames() {
+			return RecycledNotableTags.values();
+		}
+	},
+
+
+	RESET {
+		@Override
+		public String getName() {
+			return "%s.reset";
```

Comment:
A lot of the names seem to be very generic, which requires the user to carefully provide a good prefix to be able to quickly understand the metrics when looking at them in a metrics backend. I wonder if adding some more context by adding `reactor.pool.` to the beginning of meter names will help. I also wonder if allowing a customizable prefix is necessary in this case. I assume the motivation behind it is to allow identifying metrics for different pools, but I think that could be achieved with a `pool.name` tag or similar, in lieu of a prefix. What do you think?
Previously, we needed to add a prefix to `ExecutorServiceMetrics` in Micrometer because different libraries/users were using it with a different set of `extraTags`. If we know the set of tags is consistent (because we don't accept arbitrary extra ones) then we can avoid this complexity, I think. /cc @jonatan-ivanov

Reply:
ok. I think this doesn't change the need for a separate module because we need to ask the user for a `MeterRegistry` anyway, but interesting point about naming meters less generically and turning the prefix into a name tag.

`metricsPrefix` is indeed intended as a way to identify distinct pools.

as a side note, what about tags naming?